crates/libcontainer/src/process/container_init_process.rs

```diff
@@ -442,7 +441,8 @@ pub fn container_init_process(
     }
 
     if proc.args().is_some() {
-        ExecutorManager::exec(spec)
+        args.executor_manager.exec(spec)?;
+        unreachable!("process image should have been replaced after exec");
```

Comment:
The image will not have been replaced if you are using wasm to execute the workload and youki will crash in that case.